src/Annotator/EntityAnnotator.php

```diff
@@ -57,15 +57,11 @@ public function annotate($path) {
 		$propertyHintMap = $helper->buildEntityPropertyHintTypeMap($schema);
 		$propertyHintMap = $this->buildExtendedEntityPropertyHintTypeMap($schema, $helper) + $propertyHintMap;
 		$propertyHintMap += $this->buildVirtualPropertyHintTypeMap($content);
+		$propertyHintMap += $helper->buildEntityAssociationHintTypeMap($schema);
 
 		$propertyHintMap = array_filter($propertyHintMap);
 
 		$annotations = $helper->propertyHints($propertyHintMap);
-		$associationHintMap = $helper->buildEntityAssociationHintTypeMap($schema);
-
-		if ($associationHintMap) {
-			$annotations = array_merge($annotations, $helper->propertyHints($associationHintMap));
```

Comment:
array_merge() often kills you in the end .. :P
I should have guessed it here, too